### PR TITLE
Fix error in [Update yaml and py file in Tavily Tool]

### LINKED ISSUE #3572: Bug with Tavily search, 3 letters or less not working as search
### Self Checks

- [X] This is only for bug report, if you would like to ask a quesion, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### Dify version

0.6.3

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

use tavily search with 3 letters or less in the search input field. 

### ✔️ Expected Behavior

.

### ❌ Actual Behavior

CleanShot 2024-04-17 at 14 17 25@2x

### PR DESCRIPTION
**Pull request by richards199999**
# Description

Fix an error in previous PR.
Fix https://github.com/langgenius/dify/issues/3572
## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml [hunk 1]
```diff
@@ -25,3 +25,142 @@ parameters:
...
+          zh_Hans: 是
+          pt_BR: Yes
+      - value: false
+        label:
+          en_US: No
+          zh_Hans: 否
+          pt_BR: No
+    default: false
+  - name: max_results
+    type: integer
```
Comment (Yeuoly) [inline api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml]: integer is not a valid parameter type

File: api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml [hunk 2]
```diff
@@ -25,3 +25,142 @@ parameters:
...
+    human_description:
+      en_US: The number of maximum search results to return
+      zh_Hans: 返回的最大搜索结果数
+      pt_BR: The number of maximum search results to return
+    form: form
+    min: 1
+    max: 20
+    default: 5
+  - name: include_domains
+    type: array
```
Comment (Yeuoly) [inline api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml]: but also, array is not supported

**Comment by Yeuoly** [comment]
I think you may need to read our Tool Integration Guide [here](https://github.com/langgenius/dify/blob/main/api/core/tools/README.md) and test it locally yourself, and there are what I saw while testing this PR.
image

**Review by Yeuoly — CHANGES_REQUESTED** [review]

**Comment by richards199999** [comment]
ahhhhhh, truly sorry for the inconvenience😭😭😭

should be more careful next time😭

File: api/core/tools/provider/builtin/tavily/tools/tavily_search.py [hunk 3]
```diff
@@ -24,87 +24,34 @@ class TavilySearch:
...
-            "max_results": max_results,
-            "search_depth": search_depth,
-            "include_domains": include_domains,
-            "exclude_domains": exclude_domains,
-            "include_answer": include_answer,
-            "include_raw_content": include_raw_content,
-            "include_images": include_images,
-        }
+        params["api_key"] = self.api_key
         response = requests.post(f"{TAVILY_API_URL}/search", json=params)
```
Comment (Yeuoly) [inline api/core/tools/provider/builtin/tavily/tools/tavily_search.py:48]: `includes_domains` and `exclude_domains` does not work here, it should be an array but got string instead, below could make sense I guess.
```python
params['exclude_domains'] = params.get('exclude_domains', '').split()
```

**Review by Yeuoly — CHANGES_REQUESTED** [review]

File: api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml [hunk 4]
```diff
@@ -25,3 +25,138 @@ parameters:
       pt_BR: used for searching
     llm_description: key words for searching
     form: llm
+  - name: search_depth
+    type: string
```
Comment (Yeuoly) [inline api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml]: Oops, there is still a type need to be change to `select`

Comment (richards199999) [reply api/core/tools/provider/builtin/tavily/tools/tavily_search.yaml]: sorry for having soooo many errors and make this PR been this long. i'm kinda busy with school stuff these days so you know😭😭😭😭
but i have change the type already!

**Review by Yeuoly — APPROVED** [review]
LGTM

**Pull request merged**